src/smart-components/workspaces/EditWorkspaceModal.tsx

```diff
@@ -56,12 +57,17 @@ const EditWorkspaceModal: React.FunctionComponent<EditWorkspaceModalProps> = ({
           component: componentTypes.TEXT_FIELD,
           validate: [
             { type: validatorTypes.REQUIRED },
-            (value: string) => {
-              if (value === initialFormData?.name) {
-                return undefined;
+            (value: string, currData: unknown | undefined) => {
+              if (isWorkspace(currData)) {
```

Comment:
We can make the `isWorkspace` a TS type guard. It will remove the necessity for typecasting in the branch.

https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types